server/raft_test.go

```diff
@@ -549,3 +549,41 @@ func TestNRGSystemClientCleanupFromAccount(t *testing.T) {
 	finish := numClients()
 	require_Equal(t, start, finish)
 }
+
+func TestNRGLeavesObserverAfterPause(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	defer c.shutdown()
+
+	rg := c.createMemRaftGroup("TEST", 3, newStateAdder)
+	rg.waitOnLeader()
+
+	n := rg[0].node().(*raft)
+
+	checkState := func(observer, pobserver bool) {
+		n.RLock()
+		defer n.RUnlock()
+		require_Equal(t, n.observer, observer)
```

Comment:
These fail will block tests since lock still held.

Reply:
It does block if you do:
```
	n.RLock()
	require_Equal(t, n.observer, observer)
	require_Equal(t, n.pobserver, pobserver)
	n.RUnlock()
```
... but not if you use `defer` as the PR does.